### DISCUSSION TITLE
Whether there is an API to determine whether lit is in development mode?

### DISCUSSION THREAD

**Discussion by nmnmcc** [lit/lit#4653] [Q&A] [answered]
?

**Comment by drk-mtr** [comment]
My understanding is that it isn't lit that is in development mode or not, it's the server you're using.

For example if you're using vite as your development server, you could use `import.meta.env.MODE` https://vitejs.dev/guide/env-and-mode

**Comment by augustjk** [comment, marked as answer]
Lit uses node's [conditional export](https://nodejs.org/api/packages.html#conditional-exports) for development builds. So like the answer from `drk-mtr`, it depends on your environment or build tool whether the "development" condition is chosen on import, and those build tools may have ways for you to detect the mode they're running in.

An approximate runtime check for Lit could be something like checking whether `LitElement.enableWarning` is defined, as it'll only be defined in development mode. It's not a published API so it could change in the future though.

**Comment by nmnmcc** [comment]
Thank you!
